src/download-manager.js

```diff
@@ -37,205 +38,102 @@ const browserManager = require('./browser-manager.js');
-              process.platform));
-        }
-
-        return reject(new Error('Unable to find appropriate download for ' +
-          'this environment.'));
-      });
+  _getStorage() {
+    storage.initSync({
+      dir: path.join(application.getInstallDirectory(), 'database/')
     });
+
+    return storage;
   }
 
   /**
-   * This method retrieves the Firefox Gecko driver, and makes it
-   * executable. It's installed to the directory defined via the
-   * application get/setInstallDir() and it's added to the processes
-   * PATH for use by selenium-webdriver.
-   * @return {Promise}              Resolves when the driver is downloaded
+   * This method will download a browser if it is needed (i.e. can't be found
+   * in the usual system location or in the install directory).
+   * @param  {String} browserId This is the Selenium ID of the browser you wish
+   *                            to download ('chrome', 'firefox', 'opera').
+   * @param  {String} release   This downloads the browser on a particular track
+   *                            and can be 'stable', 'beta' or 'unstable'
+   * @param  {int} [expirationInHours=24] This is how long until a browser
+   *                             download is regarded and expired and Should
+   *                             be updated. A value of 0 will force a download.
+   *         										 If you want to install the browser regardless
+   *                             of any existing installs of the process, pass
+   *                             in true.
+   * @return {Promise}           Promise resolves once the browser has been
+   *                             downloaded and ready for use.
    */
-  downloadFirefoxDriver() {
-    let geckoDriverPath = path.join(
-      application.getInstallDirectory(), 'geckodriver');
+  downloadBrowser(browserId, release, expirationInHours) {
+    let installDir = application.getInstallDirectory();
+
+    const storage = this._getStorage();
+    const storageKey = `${browserId}:${release}`;
 
     return new Promise((resolve, reject) => {
-      mkdirp(geckoDriverPath, err => {
+      storage.getItem(storageKey, (err, value) => {
         if (err) {
-          return reject(err);
+          reject(err);
+          return;
         }
-        resolve();
+
+        resolve(value);
       });
     })
-    .then(() => {
-      return this._getFirefoxDriverDownloadURL();
-    })
-    .then(downloadInfo => {
-      return new Promise((resolve, reject) => {
-        // Make sure we know what we are dealing with
-        const expectExtension = '.tar.gz';
-        if (downloadInfo.name.indexOf(expectExtension) !==
-          (downloadInfo.name.length - expectExtension.length)) {
-          return reject(new Error(`Unexpected file extension for ` +
-            `${downloadInfo.name}`));
+    .then(lastBrowserUpdate => {
+      if (lastBrowserUpdate) {
+        if (typeof expirationInHours === 'undefined') {
+          expirationInHours = 24;
```

Comment:
Instead of hardcoding `24` here, can you expose it as either something in `module.exports` or (which might be cleaner), expose it as a `static` property of the `DownloadManager` class? That way you can make reference to that value directly in your tests, instead of having to keep the test value in sync with the constant here.